packages/node/src/fork-dev-server.ts

```diff
@@ -0,0 +1,102 @@
+import once from '@tootallnate/once';
+import { cloneEnv, Config, Meta } from '@vercel/build-utils';
+import { ChildProcess, fork, ForkOptions } from 'child_process';
+import { pathToFileURL } from 'url';
+import { join } from 'path';
+
+export function forkDevServer(options: {
+  tsConfig: any;
+  config: Config;
+  maybeTranspile: boolean;
+  workPath: string | undefined;
+  isTypeScript: boolean;
+  isEsm: boolean;
+  require_: NodeRequire;
+  entrypoint: string;
+  meta: Meta;
+
+  /**
+   * A path to the dev-server path. This is used in tests.
+   */
+  devServerPath?: string;
+}) {
+  let nodeOptions = process.env.NODE_OPTIONS;
+  const tsNodePath = options.require_.resolve('ts-node');
+  const esmLoader = pathToFileURL(join(tsNodePath, '..', '..', 'esm.mjs'));
```

Comment:
**question (blocking):** what does this fix, exactly?

I tried latest `vc` on some example functions that were written in ESM/MJS style on Windows, but they worked fine. How do I reproduce the issue?

Reply:
It must be Windows + TypeScript for it to happen. If I understand correctly, when running on ESM, it is expected to have the loader in the `file://` format. Unix paths are allowed too, but full windows path is being parsed wrongly, where the drive is the protocol. So the error says “c: is not a valid protocol” 😩

makes sense?

Reply:
Again the first commits were to reproduce. Once I had a single failing test I fixed it :) so this ensures this very line fixes the issue

Reply:
I can't run this repo on windows for some reason (looking into it elsewhere), but I did create an example project that has files that look like the fixture in this PR on my windows machine. They both respond just fine with and without this PR's changes.

Has a user reported this issue or have we only seen it in Windows CI? We can certainly merge this even if it's just in Windows CI, but if it happens on user machines, I'd like to be able to reproduce to verify the fix.

Reply:
I got my windows install working and was able to reproduce this with the test. This looks good to me. Thanks!